langchain-tech-stack-hackathon/README.md

```diff
@@ -18,89 +18,120 @@ The following pre-requisites are required to get started:
 - Google Cloud Project with Vertex AI API enabled.
 - A Google account with access to the needed resources ([see below](#running-a-hackathon-event)).
 - The `gcloud` CLI tool, configured to access the hackathon GCP.
-- Python (this tutorial assumed 3.11, but other versions will work).
+- Python (this tutorial assumed 3.11, but other versions will work) (use [PyEnv](https://github.com/pyenv/pyenv) for version management).
 - Poetry (^1.6.1)
-- A [LangSmith](https://smith.langchain.com/) account (this will require sign-up - speak to @zacharysmithdatatonic for a sign-up code to gain access immediately).
 - [DirEnv](https://direnv.net/) (or another terminal secret manager).
-- A blank project, open in VSCode.
 
-## Tutorial walkthrough
+## Task 1 walkthrough
 
-0. If using a secret file (e.g.: `.envrc`), create a `.gitignore` file [like this one](https://www.toptal.com/developers/gitignore/api/direnv,python,visualstudiocode,macos), to prevent accidentally sharing your API keys.
+0. Create a `.gitignore` file with [this content](https://www.toptal.com/developers/gitignore/api/direnv,python,visualstudiocode,macos).
 
-1. Create a `.envrc` file and populate it with this template (adding in your LangSmith API key):
+1. Create a `.envrc` file and populate it with this template:
 
 ```bash
-export LANGCHAIN_TRACING_V2=true
-export LANGCHAIN_ENDPOINT=https://api.smith.langchain.com
-export LANGCHAIN_API_KEY=<your-langsmith-api-key>
-export LANGCHAIN_PROJECT=rag-google-cloud-vertexai-search  # if not specified, defaults to "default"
-
 export GOOGLE_CLOUD_PROJECT_ID=dt-gen-ai-hackathon-dev
-export DATA_STORE_ID=<your-data-store-id>
+export DATA_STORE_ID=<your-data-store-id> # replace with your data store ID for this workshop
 export MODEL_TYPE=chat-bison@001
 ```
 
-2. Open a new terminal and run these commands:
+_❗ If you don't want to use DirEnv, just run each line as a separate terminal command - the environment variables will persist till you close the terminal window_
 
-```bash
-poetry init -n --python=3.11.6 && poetry add langchain-cli
+2. Open a terminal at the root of your project:
+
+- Ensure the right Python version is activated, DirEnv can access your secrets file, and set Poetry to create the `.venv` environment folder locally (this helps if you need to tidy-up/reset the environment later):
+
+```sh
+pyenv shell 3.11.6
+direnv allow
+poetry config virtualenvs.in-project true
+```
+
+- If cloning the `pyproject.toml` and `poetry.lock` files from this repository, run these commands:
+
+```sh
+poetry install
+```
+
+- _Else_, initialise Poetry from scratch and install these requirements:
+
+```sh
+poetry install
```

Comment:
I'll make the instructions clearer - the purpose that step is _either_ to install with the `pyproject.toml` from this repo, or to install from scratch. Thoughts on this approach vs. just installing from scratch each time?

Reply:
Yes, I think that's clear. I was just saying that in the scenario where we install from scratch, we do not need to perform 'poetry install'. Instead, we only need to initialise with 'poetry init' and then add the packages, as you stated.

Maybe it will confuse users, if they do not have experience with Poetry, to understand that the second option implies that they will need to delete the .toml and the .lock, because this assumes that all users will clone the entire repository, correct?